packages/api/src/command/medical/patient/consolidated-get.ts

```diff
@@ -110,7 +130,68 @@ export async function startConsolidatedQuery({
   return progress;
 }
 
-async function getConsolidatedAndSendToCx(params: GetConsolidatedParams): Promise<void> {
+function appendProgressToProcessingQueries(
+  currentConsolidatedQueries: ConsolidatedQuery[] | undefined,
+  progress: ConsolidatedQuery
+): ConsolidatedQuery[] {
+  if (currentConsolidatedQueries) {
+    const queriesInProgress = currentConsolidatedQueries.filter(
+      query => query.status === "processing"
+    );
+
+    return [...queriesInProgress, progress];
+  }
+
+  return [progress];
+}
+
+export function getCurrentConsolidatedProgress(
+  consolidatedQueriesProgress: ConsolidatedQuery[] | undefined,
+  queryParams: GetConsolidatedFilters
+): ConsolidatedQuery | undefined {
+  if (!consolidatedQueriesProgress) return undefined;
+
+  const { resources, dateFrom, dateTo, conversionType } = queryParams;
+
+  for (const progress of consolidatedQueriesProgress) {
+    const isSameResources = getIsSameResources(resources, progress.resources);
+    const isSameDateFrom = progress.dateFrom === dateFrom;
+    const isSameDateTo = progress.dateTo === dateTo;
+    const isSameConversionType = progress.conversionType === conversionType;
+    const isProcessing = progress.status === "processing";
+
+    if (isSameResources && isSameDateFrom && isSameDateTo && isSameConversionType && isProcessing) {
+      return progress;
+    }
+  }
+}
+
+export function getIsSameResources(
+  queryResources: ResourceTypeForConsolidation[] | undefined,
+  currentResources: ResourceTypeForConsolidation[] | undefined
+): boolean {
+  const haveSameLength = queryResources?.length === currentResources?.length;
+  const intersectedResources = intersection(queryResources, currentResources);
+  const usingAllQueryResources = queryResources?.length === intersectedResources.length;
+
+  const areQueryResourcesSearchableByPatient =
+    intersection(queryResources, resourcesSearchableByPatient).length ===
+    resourcesSearchableByPatient.length;
+  const areQueryResourcesEmpty = !queryResources || queryResources.length === 0;
+
+  const isCurrentProgressSearchableByPatient =
+    intersection(currentResources, resourcesSearchableByPatient).length ===
+    resourcesSearchableByPatient.length;
+  const isCurrentProgressEmpty = !currentResources || currentResources.length === 0;
+
+  return (
+    (haveSameLength && usingAllQueryResources) ||
+    (isCurrentProgressEmpty && areQueryResourcesSearchableByPatient) ||
+    (areQueryResourcesEmpty && isCurrentProgressSearchableByPatient)
```

Comment:
I don't understand why we need these.

This is saying that we consider the resources the same if:
- the existing query didn't have any resources but the new one is querying by all resources searchable by patient; or
- the new query is not filtering by patient but the previous one was querying by all resources searchable by patient.

Reply:
Because empty array is the same as all the filters.

Reply:
Gotcha, I see now. The naming was confusing me: `areQueryResourcesSearchableByPatient` doesn't give me the sense that it means "user included all resources searchable by patient".